File: src/client/Models/Option.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace client.Models
{
    public class Option : Instrument
    {
        public Proto.OptionType OptionType { get; set; }
        public Proto.ExerciseType ExerciseType { get; set; }
        public Proto.SettlementType SettlementType { get; set; }
        public double Strike { get; set; }
    }
}

File: src/server/model/Option.cpp
#include "Option.h"
// #include "Message.h"

// std::shared_ptr<Proto::Instrument> Option::Serialize() const {
//   auto inst = Message::NewProto<Proto::Instrument>();
//   inst->set_id(id_);
//   inst->set_symbol(symbol_);
//   inst->set_exchange(exchange_);
//   inst->set_type(type_);
//   inst->set_currency(currency_);
//   inst->set_tick(tick_);
//   inst->set_multiplier(multiplier_);
//   if (underlying_) {
//     inst->set_underlying(underlying_->Id());
//   }
//   if (hedge_underlying_) {
//     inst->set_hedge_underlying(hedge_underlying_->Id());
//   }
//   inst->set_maturity(boost::gregorian::to_iso_string(maturity_));
//   inst->set_call_put(call_put_);
//   inst->set_exercise(exercise_type_);
//   inst->set_settlement(settlement_type_);
//   inst->set_strike(strike_);
//   return inst;
// }

void Option::Serialize(Proto::Instrument *inst) const {
  Instrument::Serialize(inst);
  inst->set_call_put(call_put_);
  inst->set_exercise(exercise_type_);
  inst->set_settlement(settlement_type_);
  inst->set_strike(strike_);
}
